unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,143 @@
+import uuid
+import pytest
+from redis import StrictRedis
+import d3a_api_client
+from d3a_api_client.redis_client_base import RedisClientBase, RedisAPIException
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+area_id = str(uuid.uuid4())
+area_uid = None
+d3a_api_client.redis_client_base.StrictRedis = MagicMock(spec=StrictRedis)
+d3a_api_client.redis_client_base.StrictRedis.publish = MagicMock(spec=StrictRedis)
+d3a_api_client.redis_client_base.StrictRedis.pubsub.psubscribe = MagicMock()
+
+
+@pytest.fixture
+def redis_client_auto_register():
+    return RedisClientBase(area_id=area_id, autoregister=False)
+
+
+class TestRedisClientBase:
+
+    def test_constructor_registry_success(self, redis_client_auto_register, pubsub_thread=None):
+        assert redis_client_auto_register.area_id == area_id
+        assert redis_client_auto_register.area_uuid == area_uid
+        assert redis_client_auto_register.is_active == False
+        assert redis_client_auto_register._subscribed_aggregator_response_cb is None
+        assert redis_client_auto_register._subscribe_to_response_channels(pubsub_thread) is None
+        assert redis_client_auto_register._blocking_command_responses == {}
+        assert redis_client_auto_register._transaction_id_buffer == []
+
+    @patch('d3a_api_client.redis_client_base.RedisClientBase._subscribe_to_response_channels')
+    def test_function_subscribe_to_response_channels_sucess(self, mock_funct):
+        f = mock_funct()
```

Comment:
@karthikyadav3 You are calling the patched function directly and do not create an instance of `RedisClientBase` in order to call the `__init__` . Here you call the function `_subscribe_to_response_channels` again and check whether it was called. This is not the goal of the unitest

Reply:
Do I need a call like this: 
``` @patch('d3a_api_client.redis_client_base.RedisClientBase')
    def test_function_subscribe_to_response_channels_sucess(self, mock_funct):
        f = mock_funct()
        mock_funct._subscribe_to_response_channel.sassert_called_once()
```

Reply:
I think so.